### PR TITLE
Add nix flake and nixos service.

### PR DESCRIPTION
**Pull request by jbgi**

### CODE REVIEW

**Review by KtorZ — COMMENTED** [review]
Sweet :pray: 
This makes sense to me, although I am not very familiar with Flake & al.

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,106 @@
+{
```
[the remark this answers is not in the document]
Comment (KtorZ) [reply flake.nix:1]: Actually, found this: https://www.tweag.io/blog/2020-05-25-flakes/

**Comment by KtorZ** [comment]
Hmm. It seems like this is also now breaking caching with cachix in the CI job: 

https://github.com/KtorZ/cardano-ogmios/pull/56/checks?check_run_id=2586125054

:thinking: ... I'll try to fiddle around with it.

**Comment by jbgi** [comment]
Thanks for fixing!

**Pull request closed without merging**